src/Tasks/SystemState.cs

```diff
@@ -366,15 +366,10 @@ private FileState ComputeFileStateFromCachesAndDisk(string path)
             // If the process-wide cache contains an up-to-date FileState, always use it
             if (isProcessFileStateUpToDate)
             {
-                // If a FileState already exists in this instance cache due to deserialization, remove it;
-                // another instance has taken responsibility for serialization, and keeping this would
-                // result in multiple instances serializing the same data to disk
-                if (isCachedInInstance)
+                if (!isInstanceFileStateUpToDate)
                 {
-                    instanceLocalFileStateCache.Remove(path);
-                    isDirty = true;
+                    instanceLocalFileStateCache[path] = cachedProcessFileState;
```

Comment:
Please set `isDirty = true;` here to force its cache file saving.

Reply:
```suggestion
                    instanceLocalFileStateCache[path] = cachedProcessFileState;
                    isDirty = true;
```